lib/ansible/modules/storage/netapp/na_ontap_user_role.py

```diff
@@ -0,0 +1,211 @@
+#!/usr/bin/python
+
+# (c) 2018, NetApp, Inc
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+
+DOCUMENTATION = '''
+
+module: na_ontap_user_role
+
+short_description: useradmin configuration and management
+extends_documentation_fragment:
+    - netapp.na_ontap
+version_added: '2.6'
+author: Sumit Kumar (sumit4@netapp.com)
+
+description:
+- Create or destroy user roles
+
+options:
+
+  state:
+    description:
+    - Whether the specified user should exist or not.
+    required: true
+    choices: ['present', 'absent']
+    default: 'present'
+
+  name:
+    description:
+    - The name of the role to manage.
+    required: true
+
+  command_directory_name:
+    description:
+    - The command or command directory to which the role has an access.
+    required: true
```

Comment:
only required when state==present?

Reply:
We'll look in to this for the next release.